include/emp/prefab/ControlPanel.hpp

```diff
@@ -0,0 +1,331 @@
+#ifndef EMP_CONTROL_PANEL_HPP
+#define EMP_CONTROL_PANEL_HPP
+
+#include "emp/base/optional.hpp"
+#include "emp/base/vector.hpp"
+
+#include "emp/prefab/ButtonGroup.hpp"
+#include "emp/prefab/FontAwesomeIcon.hpp"
+#include "emp/prefab/ToggleButtonGroup.hpp"
+
+#include "emp/tools/string_utils.hpp"
+
+#include "emp/web/Animate.hpp"
+#include "emp/web/Element.hpp"
+#include "emp/web/Div.hpp"
+
+namespace emp::prefab {
+
+  namespace internal {
+    using checker_func_t = std::function<bool(const web::Animate &)>;
+    /**
+     * Shared pointer held by instances of ControlPanel class representing
+     * the same conceptual ControlPanel DOM object.
+     * Contains state that should persist while ControlPanel DOM object
+     * persists.
+     */
+    class ControlPanelInfo : public web::internal::DivInfo {
+
+      // The unit for rate of refresh
+      std::string refresh_unit;
+
+      // A list of refresh rates with current value associated with current unit
+      std::map<std::string, int> refresh_rates{
+        {"MILLISECONDS", 100}, {"FRAMES", 5}
+      };
+
+      // Map units to cool refresh checkers
+      const std::map<std::string, const checker_func_t> refresh_checkers{
+        { "MILLISECONDS",
+          [elapsed_milliseconds = 0, this]
+          (const web::Animate & anim) mutable {
+            int rate = this->refresh_rates[this->refresh_unit];
+            elapsed_milliseconds += anim.GetStepTime();
+            if (elapsed_milliseconds > rate) {
+              elapsed_milliseconds -= rate;
+              if (elapsed_milliseconds > rate) elapsed_milliseconds = 0;
+              return true;
+              // If this is weird check out explanation of pattern in ReadoutPanel
+            }
+            return false;
+          }},
+        { "FRAMES",
+          [this](const web::Animate & anim) {
+            return anim.GetFrameCount() % this->refresh_rates[this->refresh_unit];
+          }
+        }
+      };
+
+      // The current redraw checker function
+      checker_func_t do_redraw;
+
+      // A list of widget that should be redrawn when do_redraw return true
+      emp::vector<web::Widget> refresh_list;
+
+      // A function to run every frame (as fast as possible)
+      std::function<void()> simulation;
+
+      public:
+      /**
+       * Construct a shared pointer to manage ControlPanel state.
+       *
+       * @param in_id HTML ID of ConfigPanel div
+       */
+      ControlPanelInfo(const std::string & in_id="")
+      : DivInfo(in_id),
+      refresh_unit("MILLISECONDS"),
+      do_redraw(refresh_checkers.at(refresh_unit)),
+      simulation([](){ ; })  { ; }
+
+      /**
+       * Get a reference to the redraw checker function
+       * @return a redraw checker (void(const Animate &) function)
+       */
+      const checker_func_t & GetRedrawChecker() const {
+        return do_redraw;
+      }
+
+      /**
+       * Set the simulation for this control panel
```

Comment:
Explain that by simulation you mean a void callback that can be used to advance the simulation one frame